ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FooSubmission.scala

```diff
@@ -14,6 +15,8 @@ class FooSubmission(
     submissionConfig: FooSubmissionConfig,
     allocatedParties: AllocatedParties,
     names: Names,
+    partySelectingRandomnessProvider: RandomnessProvider = RandomnessProvider.Default,
```

Comment:
It is not clear to me why we need several distinctive randomness providers

Reply:
We could do with just one provider but then the tests could be more easily broken by changes in the code, even benign refactors, when party-selecting and consuming-event-generation were swapped in the code/execution order.